llvm/test/CodeGen/RISCV/cheri/isav9-cap-to-ptr.ll

```diff
@@ -0,0 +1,61 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+;; The CToPtr instruction is not part of ISAv9, check that we emit the
+;; optimized form of `x.tag ? x.addr : 0` instead.
+; RUN: %riscv64_cheri_purecap_llc < %s | FileCheck %s --check-prefix=ISAV8
+; RUN: %riscv64_cheri_purecap_llc -mattr=+xcheri-no-v8-compat < %s | FileCheck %s --check-prefix=ISAV9
+;; Also emit a .o file since verifyInstructionPredicates() is only called for binary output.
+; RUN: %riscv64_cheri_purecap_llc -mattr=+xcheri-no-v8-compat -filetype=obj -o /dev/null < %s
+
+;; (int_cheri_cap_from_ptr x, y) -> y == 0 ? null : csetaddr(x, y)
+define internal i64 @cap_to_ptr(i64 addrspace(200)* %dst, i8 addrspace(200)* %auth, i8 addrspace(200)* %cap) nounwind {
```

Comment:
Why internal over dso_local? internal is like static, so this code is dead if you put it through opt.

Reply:
I believe that was a copy-past from some other test - will update to use dso_local.